### ISSUE TITLE
suggestion: coreutils by default

### ISSUE THREAD

**Issue by fornwall** [termux/termux-packages#178]
_From @InsideBall on March 12, 2016 2:19_

I suggest that this app installs coreutils by default, and sets path correctly so they are used by default, by default.

I suggest this because the default shell is bash. This is pretty awesome. If you set coreutils by default, most bash scripts scripts written for GNU + Linux will run. Also, because, why not. These are Ghz+ multi-core devices we run on, with gigabytes of internal storage, no need for busybox.

I also suggest adding the net-tools package for networking.

_Copied from original issue: termux/termux-app#62_

**Comment by its-pointless** [comment]
I think the splash screen should strongly hint you should install coreutils. Getting a somewhat useful system and not using much data might be useful in of itself for some people though.

**Comment by smemsh** [comment]
I have to agree, I had just had a bunch of issues bringing in some scripts from another platform because things like `id` and `hostname` were not present, I ended up building my own busybox with more stuff in it instead because I had no idea to install coreutils until I came to the website much later (which does suggest to install coreutils).  I assumed there was some platform-specific reason busybox had to be used...

I had the following issues with things which I thought were fairly standard, after installing `termux`:

- no `egrep` (broke several configure scripts)
- no `id`, `tty`, `uname`, `hostname` (broke some shell init scripts)
- `pgrep does not have `-u` (i figured out, no procps!!!)
- no `mktemp` (breaks many scripts)

there should probably also be `tar`, `sed`, `grep`, `procps`, `coreutils`, `bash`, `bsdutils`, `diffutils`, `findutils`, `gzip`, `hostname`, `util-linux` and `gawk` if they aren't there already, these are fairly standard base utilities any linux install should have.

**Comment by fornwall** [comment]
@smemsh `mktemp` and `uname` should be available by default from busybox, are you sure that they didn't exist for you?

I've now also added `egrep`, `hostname`, `id`,  and `tty` to the busybox configuration, so more things should now work out of the box.

Perhaps a virtual package like `linux-base` or something can be created, which depends on `coreutils` and other packages that are considered standard for a typical Linux installation?

**Comment by smemsh** [comment]
yes, you're right about `uname` and `mktemp`; I think `.../usr/bin/applets` was accidentally removed from my path (leaving only `.../usr/bin`) during my test.  I had force-set it but only after the time it was used.

busybox apps seem quite limited, for example I just tried "wget https://github.com/smemsh.keys" and got "invalid argument" but it worked fine after installing the "real" wget.  Several other examples of this, missing flags, ignored flags, etc.  I agree that busybox should probably just be skipped in favor of a minimal set of the canonical utilities later.  However @its-pointless is right that this would add considerably to the initial bootstrap size, which might not be useful to everyone.

The virtual package idea is a good one; the list could really be the ones from e.g. `debootstrap --variant=minbase` but without the ones that don't make sense on a non-root system.  However, rather than duplicate that whole recursive dependency tree and keep it up to date, it might be easier to use it as-is but provide fake packages / stubs for the ones that don't make sense in a user-based environment with no root.

Then again, maybe just a short list of packages is easier and change it as needed... especially if planning to always keep it as a restricted functionality user app with no plans to ever support e.g. chrooting into `.../com.termux/files` on rooted phones.

**Comment by leoj3n** [comment]
@smemsh Looks like busybox's `wget` doesn't support HTTPS but HTTP and FTP are supported according to help text.

**Comment by sdrausty** [comment]
@fornwall perhaps we could implement something along the line of this option `pacman -S base-devel` in Arch Linux in Termux PRoot which Installs the developer's base?  This would keep the bootstrap download as small as possible, while at the same time giving developers the option to install what they want for their systems just a few keystrokes.  

![screenshot_20171205-170813](https://user-images.githubusercontent.com/27742457/33633516-22d42aaa-d9df-11e7-9f67-7988b88aafad.png)

**Comment by ncoop** [comment]
I disagree with the OP request. If you want something, pkg install it for yourself. Some users like having choice and don't want bloat. Enabling more of the BusyBox tools is a good compromise.

Even bash being a required package is problematic. BusyBox has a perfectly good implementation of ash. What's wrong with typing `pkg install bash` one time?

**Comment by xeffyr** [comment]
> What's wrong with typing `pkg install bash` one time?

Bash is such big package so your device can't handle it ?
**Bash is standard shell for many Linux distributions.**

**Comment by xeffyr** [comment]
Closing in favor of https://github.com/termux/termux-packages/issues/4070.
